### PR TITLE
Add ability to clean the DB in DevUI

### LINKED ISSUE #583: Consider offering a way to reset DB at Hibernate ORM level for testing
reset would rerun the DDL statements and import.sql file.

This is useful if you want a test to be fully isolated from another data wise.

### LINKED ISSUE #19303: Button to reset the database state from the Quarkus Dev UI for Hibernate ORM
### Description

See https://github.com/quarkusio/quarkus/pull/16764#pullrequestreview-648431733

> a button to reset the DB state (actually runs the drop script, followed by the create and optionally the import.sql if there's any)

CC @Sanne  @TomasHofman

### Implementation ideas

I think we would definitely need a pop-up with a warning, even if it's just development mode. Some people work with pre-initialized databases which take a long time to initialize, and they would be very annoyed if they dropped it by mistake.

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #583
Fixes #19303

### CODE REVIEW

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 1]
```diff
@@ -471,6 +471,11 @@ TIP: It is possible to read annotations from the test class or method to control
 WARNING: While it is possible to use JUnit Jupiter callback interfaces like `BeforeEachCallback`, you might run into classloading issues because Quarkus has
          to run tests in a custom classloader which JUnit is not aware of.
 
+== Reset the Database after tests
+
+You can use the `@io.quarkus.test.ResetDatabase` annotation to reset the database after a test has run. This will drop the database,
+and recreate the schema. Quarkus can use Liqibase, FlyWay or Hibernate ORM to reset the schema, depending on what is configured.
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:477]:
```suggestion
and recreate the schema. Quarkus can use Liquibase, Flyway or Hibernate ORM to reset the schema, depending on what is configured.
```

**Review by gsmet — DISMISSED** [review]
Let's wait for https://github.com/quarkusio/quarkus/pull/16764#issuecomment-895777440 to land before doing something with this one.

It will probably require a bit more work to be integrated with @TomasHofman 's work.

/cc @TomasHofman @yrodiere

**Comment by stuartwdouglas** [comment]
@gsmet should be all good now.

**Review by gsmet — CHANGES_REQUESTED** [review]
@stuartwdouglas I rebased and pushed a few small changes to improve the UI.

BUT... there is something very weird, I'm trying it with the `hibernate-orm-quickstart` and when I do:
- mvn clean quarkus:dev
- then `q`

I get:
```
2021-08-17 16:35:05,135 WARN  [io.qua.hib.orm.run.JPAConfig] (Shutdown thread) Unable to close the EntityManagerFactory: io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit@23937bf7: java.lang.IllegalStateException: EntityManagerFactory is closed
	at org.hibernate.internal.SessionFactoryImpl.validateNotClosed(SessionFactoryImpl.java:539)
	at org.hibernate.internal.SessionFactoryImpl.getProperties(SessionFactoryImpl.java:533)
	at io.quarkus.hibernate.orm.runtime.schema.SchemaManagementIntegrator.defaultName(SchemaManagementIntegrator.java:61)
	at io.quarkus.hibernate.orm.runtime.schema.SchemaManagementIntegrator.disintegrate(SchemaManagementIntegrator.java:45)
	at org.hibernate.internal.SessionFactoryImpl$1IntegratorObserver.sessionFactoryClosed(SessionFactoryImpl.java:283)
	at org.hibernate.internal.SessionFactoryObserverChain.sessionFactoryClosed(SessionFactoryObserverChain.java:61)
	at org.hibernate.internal.SessionFactoryImpl.close(SessionFactoryImpl.java:846)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.close(JPAConfig.java:161)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.destroy(JPAConfig.java:120)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Subclass.destroy$$superforward1(JPAConfig_Subclass.zig:282)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Subclass$$function$$5.apply(JPAConfig_Subclass$$function$$5.zig:32)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:51)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(InvocationInterceptor_Bean.zig:521)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Subclass.destroy(JPAConfig_Subclass.zig:711)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Observer_destroy_4d75ae0cbe7e61a766b92c59cd8bbdeb2c1debf7.notify(JPAConfig_Observer_destroy_4d75ae0cbe7e61a766b92c59cd8bbdeb2c1debf7.zig:189)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:307)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:295)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:237)
	at io.quarkus.arc.impl.ArcContainerImpl.shutdown(ArcContainerImpl.java:359)
	at io.quarkus.arc.Arc.shutdown(Arc.java:49)
	at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:44)
	at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:83)
	at io.quarkus.runtime.StartupContext.close(StartupContext.java:72)
	at io.quarkus.runner.ApplicationImpl.doStop(ApplicationImpl.zig:924)
	at io.quarkus.runtime.Application.stop(Application.java:203)
	at io.quarkus.runtime.Application.stop(Application.java:155)
	at io.quarkus.runtime.ApplicationLifecycleManager$ShutdownHookThread.run(ApplicationLifecycleManager.java:401)
```

I tried to reproduce it with latest `main` and latest `main` do not seem to have the same issue (I tried 3 times each and I get the error every time with your branch and never with `main`).

No idea how your code could trigger this though...

**Comment by stuartwdouglas** [comment]
@gsmet looks like something I added for HR caused issues with blocking hibernate. Should be fixed now.

**Comment by stuartwdouglas** [comment]
@gsmet I think I accidentally pushed this over your changes

**Comment by stuartwdouglas** [comment]
Never mind, I realized I could manually fetch the commit and apply it from the history on this page.

**Comment by gsmet** [comment]
OK, cool, let me retest this.

**Review by gsmet — APPROVED** [review]
Looks good now!

**Pull request merged**